torch/_inductor/codegen/triton.py

```diff
@@ -807,7 +807,9 @@ def indexing(
             if tree.prefix.upper() not in config.triton.max_block:
                 continue
             max_block = config.triton.max_block[tree.prefix.upper()]
-            if tree.numel == 1 or tree.numel % max_block == 0:
+            if V.graph.sizevars.maybe_guard_equals(
+                tree.numel, 1
+            ) or V.graph.sizevars.maybe_guard_multiple_of(tree.numel, max_block):
```

Comment:
Hm, the way this guard is implemented it pretty much becomes a static size guard on tree numel:
```
self.guard_equals(multiple * denominator, numerator)
```
@chillee is there a better way of implementing it?

Reply:
I had a fix for it here https://github.com/pytorch/pytorch/pull/87709/files#diff-20cf953dd294aee73063e638ec8985dda70d3cb4a43cf76924849dd0d3871a75R345, but I haven't landed the pr upon your recommendation haa

Reply:
lol I don't remember anything about this guard, I guess we could sneak the guard part into this PR. The storage offset is a bit sketchy, do we still need it?

Reply:
ugh, we might, but i havent pressed on it, in part because it complicates cudagraphs lol. but yea regardless, we could sneak the guard part in here yea

Reply:
also nvm, I was think of the rand recomputation thing which was the pr not to land, not storage offset

Reply:
Sounds good. I added the guard part. @eellison, this is all that was needed, correct?

Reply:
Cool, should be good to land

Reply:
oops, landed the same fix here: https://github.com/pytorch/pytorch/pull/92320/files#diff-20cf953dd294aee73063e638ec8985dda70d3cb4a43cf76924849dd0d3871a75L354

Reply:
That's fine, #91254 was reverted anyway